iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -425,6 +426,16 @@ public boolean canProvideBasicStatistics() {
     return true;
   }
 
+  @Override
+  public boolean canProvidePartitionStatistics(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
+    Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
+    if (table.currentSnapshot() != null) {
```

Comment:
We can also check branch/tag here.

Reply:
@zhangbutao thanks for checking. I am again stepping on https://github.com/apache/hive/pull/5215
Could you please address the review comments so we can merge it and reuse here?